### PR TITLE
feature: prompt embed caching

### PR DESCRIPTION
**Pull request by JoeGaffney**

### CODE REVIEW

File: workers/common/prompt_caching.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        result = original_encode_prompt(*args, **kwargs)
+
+        add_prompt_cache(cache_key, result)
+
+        return result
+
+    # Monkey patch the instance method
+    pipeline.encode_prompt = wrapped_encode_prompt
+    pipeline._prompt_cache_enabled = True
+    return pipeline
```
[the remark this answers is not in the document]
Comment (JoeGaffney) [reply workers/common/prompt_caching.py:110]: its never threaded

File: workers/common/prompt_caching.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+import torch
+from diffusers import DiffusionPipeline
+
+from common.logger import logger
+
+# Global cache for prompt embeddings
+# Key is (pipeline_type, args, kwargs)
+GLOBAL_PROMPT_CACHE = OrderedDict()
+MAX_PROMPT_CACHE_SIZE = 64
```
[the remark this answers is not in the document]
Comment (JoeGaffney) [reply workers/common/prompt_caching.py:13]: fine for now prompts are allready limited to a certain token range

File: workers/common/prompt_caching.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoeGaffney) [reply workers/common/prompt_caching.py:110]: it inherently covered by seed tests
